koboldcpp.py

```diff
@@ -3071,39 +3066,41 @@ def display_help():
             wb.open("https://github.com/LostRuins/koboldcpp/wiki")
         except:
             print("Cannot launch help in browser.")
+
     def display_updates():
         try:
-            import webbrowser as wb
-            wb.open("https://github.com/LostRuins/koboldcpp/releases/latest")
+            import webbrowser
+            webbrowser.open("https://github.com/LostRuins/koboldcpp/releases/latest")
         except:
             print("Cannot launch updates in browser.")
-
-    ctk.CTkButton(tabs , text = "Launch", fg_color="#2f8d3c", hover_color="#2faa3c", command = guilaunch, width=80, height = 35 ).grid(row=1,column=1, stick="se", padx= 25, pady=5)
-
-    ctk.CTkButton(tabs , text = "Update", fg_color="#9900cc", hover_color="#aa11dd", command = display_updates, width=90, height = 35 ).grid(row=1,column=0, stick="sw", padx= 5, pady=5)
-    ctk.CTkButton(tabs , text = "Save", fg_color="#084a66", hover_color="#085a88", command = save_config_gui, width=60, height = 35 ).grid(row=1,column=1, stick="sw", padx= 5, pady=5)
-    ctk.CTkButton(tabs , text = "Load", fg_color="#084a66", hover_color="#085a88", command = load_config_gui, width=60, height = 35 ).grid(row=1,column=1, stick="sw", padx= 70, pady=5)
-    ctk.CTkButton(tabs , text = "Help", fg_color="#992222", hover_color="#bb3333", command = display_help, width=60, height = 35 ).grid(row=1,column=1, stick="sw", padx= 135, pady=5)
-
-    # start a thread that tries to get actual gpu names and layer counts
+    
+    buttons = [
+        ("Launch", "#2f8d3c", "#2faa3c", guilaunch, 80, 1, 1, "se", 25),
+        ("Update", "#9900cc", "#aa11dd", display_updates, 90, 1, 0, "sw", 5),
+        ("Save", "#084a66", "#085a88", save_config_gui, 60, 1, 1, "sw", 5),
+        ("Load", "#084a66", "#085a88", load_config_gui, 60, 1, 1, "sw", 70),
+        ("Help", "#992222", "#bb3333", display_help, 60, 1, 1, "sw", 135)
+    ]
+    
+    for text, fg, hover, command, width, row, col, stick, padx in buttons:
```

Comment:
Doing it this way is significantly less readable. In the original code, it's very clear what width means, e.g. 80 is referring to width. Now, by splitting it into an array of unnamed values, you even have a width but not the height. You have the x padding but not the y padding. It becomes super confusing.